skyportal/handlers/api/observation.py

```diff
@@ -0,0 +1,360 @@
+                        == ExecutedObservation.instrument_field_id,
+                        ExecutedObservation.obstime >= start_date,
+                        ExecutedObservation.obstime <= end_date,
+                    )
+                    .subquery()
+                )
+                prob = sa.func.sum(
+                    LocalizationTile.probdensity
+                    * (union.columns.healpix * LocalizationTile.healpix).area
+                )
+                query = sa.select(prob).filter(
+                    LocalizationTile.localization_id == localization.id,
+                    union.columns.healpix.overlaps(LocalizationTile.healpix),
+                )
+                intprob = DBSession().execute(query).scalar_one()
+
+        observations = obs_query.all()
+
+        if return_probability:
+            data = {
+                "observations": [o.to_dict() for o in observations],
+                "probability": intprob,
+            }
+        else:
+            data = observations
+
+        return self.success(data=data)
+
+
+def add_tiles(instrument_id, instrument_name, obstable):
+    session = Session()
+    try:
+        bands = {1: 'g', 2: 'r', 3: 'i', 4: 'z', 5: 'J'}
+
+        observations = []
+        obs_grouped_by_exp = obstable.groupby('expid')
+        for group_name, df_group in obs_grouped_by_exp:
+            field_id = int(df_group["field"].median())
+            field = (
+                session.query(InstrumentField)
+                .filter_by(instrument_id=instrument_id, field_id=field_id)
+                .first()
+            )
+            if field is None:
+                return log(
+                    f"Unable to add observations for instrument {instrument_id}: Missing field {field_id}"
+                )
+
+            if instrument_name == "ZTF":
+                processed_fraction = len(df_group["field"]) / 64.0
+            else:
+                processed_fraction = 1
+
+            observations.append(
+                ExecutedObservation(
+                    instrument_id=instrument_id,
+                    observation_id=int(df_group["expid"].median()),
+                    instrument_field_id=field.id,
+                    obstime=Time(df_group["jd"].median(), format='jd').datetime,
+                    seeing=df_group["sciinpseeing"].median(),
+                    limmag=df_group["diffmaglim"].median(),
+                    exposure_time=df_group["exptime"].median(),
+                    filt=bands[int(df_group["fid"].median())],
+                    processed_fraction=processed_fraction,
+                )
+            )
+        session.add_all(observations)
+        session.commit()
+
+        return log(f"Successfully add observations for instrument {instrument_id}")
+    except Exception as e:
+        return log(f"Unable to add observations for instrument {instrument_id}: {e}")
+    finally:
+        Session.remove()
```

Comment:
I would add a DELETE endpoint as well. It may seem useless right now, but as soon as you start playing around with it you'll make mistakes and want a way to delete observations. 
You may also want to return the internal IDs of each row in the table (both in the return value for a successfull POST and when calling GET), so you can use these IDs to issue a delete using an array or a single ID (maybe both options?)

Reply:
Added a delete and for now just delete by observation_id, which I think is ok.